src/components/metadata/templates/Listing.js

```diff
@@ -0,0 +1,275 @@
+                    id={build(dialogID, ids.BUTTONS.CONFIRM)}
+                    color="primary"
+                    variant="contained"
+                    onClick={() =>
+                        selectedTemplateId &&
+                        onSelectTemplate(selectedTemplateId)
+                    }
+                >
+                    {t("common:select")}
+                </Button>,
+            ]}
+        >
+            <TableContainer component={Paper}>
+                <Table>
+                    {isFetching ? (
+                        <TableLoading
+                            baseId={tableID}
+                            numColumns={columnData?.length}
+                            numRows={5}
+                        />
+                    ) : (
+                        <TableBody>
+                            {sortedTemplates?.map((template) => (
+                                <TemplateTableRow
+                                    key={template.id}
+                                    baseId={build(tableID, template.id)}
+                                    template={template}
+                                    selected={
+                                        selectedTemplateId === template.id
+                                    }
+                                    onClick={() =>
+                                        setSelectedTemplateId(template.id)
+                                    }
+                                    onInfo={(event) => {
+                                        if (template.description) {
+                                            setTemplateInfo(
+                                                template.description
+                                            );
+                                            setTemplateInfoAnchor(
+                                                event.currentTarget
+                                            );
+                                        }
+                                    }}
+                                    onDownload={() =>
+                                        setDownloadTemplateId(template.id)
+                                    }
+                                />
+                            ))}
+                        </TableBody>
+                    )}
+                    <EnhancedTableHead
+                        baseId={tableID}
+                        columnData={columnData}
+                        order={order}
+                        orderBy={orderBy}
+                        onRequestSort={handleRequestSort}
+                    />
+                </Table>
+            </TableContainer>
+            <TemplateInfoDisplay
+                templateInfo={templateInfo}
+                anchorEl={templateInfoAnchor}
+                onClose={() => setTemplateInfoAnchor(null)}
+            />
+        </DEDialog>
+    );
+};
+
+export default MetadataTemplateListing;
```

Comment:
I prefer this listing in a `List` view rather than in a table. Then we can eliminate an additional click for `description`(`i` button) and put description as `secondary` text prop and `download` action as `secondary` action. See examples here - https://material-ui.com/components/lists/
Regardless the buttons needs in a theme color rather than `black`.

Reply:
Good idea. I think the `List` looks better:
![Select Template dialog](https://user-images.githubusercontent.com/996408/107101476-b1e16b00-67d4-11eb-9f22-b835b346153e.png)

Mobile:
![Select Template dialog on mobile](https://user-images.githubusercontent.com/996408/107101513-d3425700-67d4-11eb-80ee-897e0c2bda56.png)